tools/gen-bootstrap.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
```

Comment:
How about `set -exu -o pipefail` for strictness and preventing mistakes?

Reply:
ok, if you say so )
BTW I prefer to check critical sections manually in order to display a good error message.